In [1]:
import numpy as np
import pandas as pd

In [2]:
df=pd.read_csv('LoanData.csv')
df

,Application_ID,Age,Gender,Education,Employment_Type,Income_Group,Credit_Group,Existing_Loan,Loan_Amount_Group,Loan_Status
0,LN0001,59,Male,Not Graduate,Other,Medium,Poor,Yes,Large,Rejected
1,LN0002,49,Female,Not Graduate,Salaried,Low,Fair,No,Large,Approved
2,LN0003,35,Female,Graduate,Business,Low,Good,No,Large,Rejected
3,LN0004,28,Male,Graduate,Business,Low,Good,No,Medium,Rejected
4,LN0005,41,Male,Graduate,Salaried,Low,Poor,No,Small,Approved
...,...,...,...,...,...,...,...,...,...,...
995,LN0996,28,Male,Not Graduate,Self-Employed,Low,Good,No,Large,Rejected
996,LN0997,48,Male,Graduate,Salaried,Low,Fair,No,Medium,Approved
997,LN0998,51,Male,Not Graduate,Salaried,Medium,Good,No,Medium,Approved
998,LN0999,29,Female,Graduate,Business,Low,Fair,No,Small,Rejected


In [3]:
df.shape

(1000, 10)

In [4]:
print('No of rows :',df.shape[0])
print('No of cols :',df.shape[1])

No of rows : 1000
No of cols : 10


In [5]:
df['Loan_Status'].unique()

array(['Rejected', 'Approved'], dtype=object)

In [6]:
df['Credit_Group'].unique()

array(['Poor', 'Fair', 'Good', 'Excellent'], dtype=object)

In [7]:
df['Gender'].unique()

array(['Male', 'Female'], dtype=object)

In [9]:
df['Employment_Type'].unique()

array(['Other', 'Salaried', 'Business', 'Self-Employed'], dtype=object)

In [10]:
df.describe(include='object')

,Application_ID,Gender,Education,Employment_Type,Income_Group,Credit_Group,Existing_Loan,Loan_Amount_Group,Loan_Status
count,1000,1000,1000,1000,1000,1000,1000,1000,1000
unique,1000,2,2,4,3,4,2,3,2
top,LN0001,Male,Graduate,Salaried,Medium,Good,No,Medium,Rejected
freq,1,631,668,459,407,405,651,434,551


In [29]:
#calculate probability of ;loan approval
p_approval=(df['Loan_Status']=='Approved').mean()  #total of 0, 1 / count of o and 1
print('Probability of getting approval is :',p_approval)

Probability of getting approval is : 0.449


In [25]:
fe=df[df['Loan_Status']=='Approved']
te=df
p=len(fe)/len(te)
p

0.449

In [27]:
#probability of getting loan rejected
(df['Loan_Status']=='Rejected').mean()  #total of 0, 1 / count of o and 1

np.float64(0.551)

In [30]:
p_Rejected=1-p_approval
print(round(p_Rejected,2))

0.55


In [31]:
print(f'Chances of getting loan rejected :{round(p_Rejected*100,2)}%')

Chances of getting loan rejected :55.1%


In [32]:
df.columns

Index(['Application_ID', 'Age', 'Gender', 'Education', 'Employment_Type',
       'Income_Group', 'Credit_Group', 'Existing_Loan', 'Loan_Amount_Group',
       'Loan_Status'],
      dtype='object')

**Que :** what is the probability of approved loan that it is a male or has good credit 

**Ans :**
- As a person can be a male and can be have a goo credit at the same time, it is a Non mutual exclusive. 

In [36]:
p_m=(df['Gender'] == 'Male').mean()
p_goodcredit = (df['Credit_Group']=='Good').mean()
p_male_with_goodcredit = ((df['Credit_Group']=='Good') & (df['Gender']=='Male')).mean()
p_male_or_goodcredit = p_m + p_goodcredit - p_male_with_goodcredit
print('probability of approved loan that it is a male or has good credit :',p_male_or_goodcredit)

probability of approved loan that it is a male or has good credit : 0.784


**Que :**
What is the probability of loan approval when you are male or female ?

In [41]:
p_m=(df['Gender'] == 'Male').mean()
p_f=(df['Gender'] == 'Female').mean()
p_male_or_female=p_m+p_f
print('probability of loan approval when you are male or female :',p_male_or_female)

probability of loan approval when you are male or female : 1.0


In [44]:
df.columns

Index(['Application_ID', 'Age', 'Gender', 'Education', 'Employment_Type',
       'Income_Group', 'Credit_Group', 'Existing_Loan', 'Loan_Amount_Group',
       'Loan_Status'],
      dtype='object')

In [47]:
# what is the probability of loan approval when the applicant has good credit
dfg=df[df['Credit_Group']=='Good']
P_LApproval=(dfg['Loan_Status']=='Approved').mean()
print('probability of loan approval when the applicant has good credit',P_LApproval)

probability of loan approval when the applicant has good credit 0.5358024691358024


In [62]:
# what is the probability that a person has excellent credit score and loan got rejected.
d_ExcellentScore=(df['Credit_Group']=='Excellent').mean()
dfe=df[df['Credit_Group']=='Excellent']
p_rejected_excellent=(dfe['Loan_Status']=='Rejected').mean()
print('probability that a person has excellent credit score and loan got rejected :',round(d_ExcellentScore*p_rejected_excellent,2))

probability that a person has excellent credit score and loan got rejected : 0.05


In [63]:
# what is the probability that a person is male and loan gets approved
p_m=(df['Gender'] == 'Male').mean()
p_r=(df['Loan_Status']=='Approved').mean()
print('probability that a person is male and loan gets approved :',round(p_m*p_r,2))

probability that a person is male and loan gets approved : 0.28


In [64]:
pd.crosstab(df['Gender'],df['Loan_Status'])

Loan_Status,Approved,Rejected
Gender,,
Female,154,215
Male,295,336


#### Frequency table for Gender

In [66]:
df['Gender'].value_counts()

Gender
Male      631
Female    369
Name: count, dtype: int64

In [67]:
df['Gender'].value_counts(normalize=True)

Gender
Male      0.631
Female    0.369
Name: proportion, dtype: float64

#### Frequency table for Credit Group

In [68]:
df['Credit_Group'].value_counts()

Credit_Group
Good         405
Fair         302
Poor         157
Excellent    136
Name: count, dtype: int64

In [70]:
df['Credit_Group'].value_counts(normalize=True)

Credit_Group
Good         0.405
Fair         0.302
Poor         0.157
Excellent    0.136
Name: proportion, dtype: float64

### Frequency Table :

1. Count the records/values 

#### Two way Frequency Table

In [72]:
pd.crosstab(df['Gender'],df['Loan_Status'])

Loan_Status,Approved,Rejected
Gender,,
Female,154,215
Male,295,336


In [73]:
pd.crosstab(df['Gender'],df['Loan_Status'],normalize=True)

Loan_Status,Approved,Rejected
Gender,,
Female,0.154,0.215
Male,0.295,0.336


- Chances of getting approved for female are 15%, while for male are 29%.
- Chances of getting rejected for female are 21%, while for male are 33%.
